File: rb/features.rb
# ThrowawayEmail SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ThrowawayEmailFeatures
  def self.make_feature(name)
    case name
    when "base"
      ThrowawayEmailBaseFeature.new
    when "test"
      ThrowawayEmailTestFeature.new
    else
      ThrowawayEmailBaseFeature.new
    end
  end
end

File: py/throwawayemail_sdk/features.py
# ThrowawayEmail SDK feature factory

from throwawayemail_sdk.feature.base_feature import ThrowawayEmailBaseFeature
from throwawayemail_sdk.feature.test_feature import ThrowawayEmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ThrowawayEmailBaseFeature(),
        "test": lambda: ThrowawayEmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
